bftengine/src/bcstatetransfer/BCStateTran.cpp

```diff
@@ -265,58 +265,64 @@ void BCStateTran::loadMetrics() {
 
-    FetchingState fs = getFetchingState();
-    LOG_INFO(STLogger, "starting state is " << stateName(fs));
+      FetchingState fs = getFetchingState();
+      LOG_INFO(STLogger, "starting state is " << stateName(fs));
 
-    if (fs == FetchingState::GettingMissingBlocks || fs == FetchingState::GettingMissingResPages) {
-      SetAllReplicasAsPreferred();
-    }
-    loadMetrics();
-  } else {
-    LOG_INFO(STLogger, "BCStateTran::init - initializing a new object");
+      if (fs == FetchingState::GettingMissingBlocks || fs == FetchingState::GettingMissingResPages) {
+        SetAllReplicasAsPreferred();
+      }
+      loadMetrics();
+    } else {
+      LOG_INFO(STLogger, "BCStateTran::init - initializing a new object");
 
-    AssertAND(maxNumOfRequiredStoredCheckpoints >= 2, maxNumOfRequiredStoredCheckpoints <= kMaxNumOfStoredCheckpoints);
+      AssertAND(maxNumOfRequiredStoredCheckpoints >= 2,
+                maxNumOfRequiredStoredCheckpoints <= kMaxNumOfStoredCheckpoints);
 
-    AssertAND(numberOfRequiredReservedPages >= 2, numberOfRequiredReservedPages <= maxNumOfReservedPages_);
-    DataStoreTransaction::Guard g(psd_->beginTransaction());
-    g.txn()->setReplicas(replicas_);
-    g.txn()->setMyReplicaId(myId_);
-    g.txn()->setFVal(fVal_);
-    g.txn()->setMaxNumOfStoredCheckpoints(maxNumOfRequiredStoredCheckpoints);
-    g.txn()->setNumberOfReservedPages(numberOfRequiredReservedPages);
-    g.txn()->setLastStoredCheckpoint(0);
-    g.txn()->setFirstStoredCheckpoint(0);
-    // TODO(TK) - check max transaction size
-    for (uint32_t i = 0; i < numberOfReservedPages_; i++)  // reset all pages
-      g.txn()->setPendingResPage(i, buffer_, sizeOfReservedPage_);
-
-    g.txn()->setIsFetchingState(false);
-    g.txn()->setFirstRequiredBlock(0);
-    g.txn()->setLastRequiredBlock(0);
-    g.txn()->setAsInitialized();
-
-    Assert(getFetchingState() == FetchingState::NotFetching);
+      AssertAND(numberOfRequiredReservedPages >= 2, numberOfRequiredReservedPages <= maxNumOfReservedPages_);
+      DataStoreTransaction::Guard g(psd_->beginTransaction());
+      g.txn()->setReplicas(replicas_);
+      g.txn()->setMyReplicaId(myId_);
+      g.txn()->setFVal(fVal_);
+      g.txn()->setMaxNumOfStoredCheckpoints(maxNumOfRequiredStoredCheckpoints);
+      g.txn()->setNumberOfReservedPages(numberOfRequiredReservedPages);
+      g.txn()->setLastStoredCheckpoint(0);
+      g.txn()->setFirstStoredCheckpoint(0);
+      // TODO(TK) - check max transaction size
+      for (uint32_t i = 0; i < numberOfReservedPages_; i++)  // reset all pages
+        g.txn()->setPendingResPage(i, buffer_, sizeOfReservedPage_);
+
+      g.txn()->setIsFetchingState(false);
+      g.txn()->setFirstRequiredBlock(0);
+      g.txn()->setLastRequiredBlock(0);
+      g.txn()->setAsInitialized();
+
+      Assert(getFetchingState() == FetchingState::NotFetching);
+    }
+  } catch (const std::exception &e) {
+    LOG_FATAL(STLogger, e.what());
+    exit(1);
   }
```

Comment:
Maybe we can `catch(...)` just in case.

Reply:
I don't think so, there's no exceptions other than std::exception thrown.

Reply:
Yep, I know, but I was saying that if somebody gets creative and does that in the future, we will not exit here.